### ISSUE TITLE
Problem with vcf output (snpeff error actually)

### ISSUE THREAD

**Issue by mjafin** [etal/cnvkit#97]
Hi Eric,
Just wanted to ask your opinion if this is a problem with cnvkit vcf output or snpeff itself:
```
[2016-04-14T12:47Z] ukapdlnx117: Error: Error while processing VCF entry (line 294) :
[2016-04-14T12:47Z] ukapdlnx117:        chr22   35006324        .       N       <DEL>   .       .       IMPRECISE;SVTYPE=DEL;END=35746318;SVLEN=-739994      GT:GQ   1/1:956
[2016-04-14T12:47Z] ukapdlnx117: java.lang.IllegalArgumentException: Comparison method violates its general contract!
[2016-04-14T12:47Z] ukapdlnx117: java.lang.IllegalArgumentException: Comparison method violates its general contract!
[2016-04-14T12:47Z] ukapdlnx117:        at java.util.ComparableTimSort.mergeHi(ComparableTimSort.java:835)
[2016-04-14T12:47Z] ukapdlnx117:        at java.util.ComparableTimSort.mergeAt(ComparableTimSort.java:453)
[2016-04-14T12:47Z] ukapdlnx117:        at java.util.ComparableTimSort.mergeForceCollapse(ComparableTimSort.java:392)
[2016-04-14T12:47Z] ukapdlnx117:        at java.util.ComparableTimSort.sort(ComparableTimSort.java:191)
[2016-04-14T12:47Z] ukapdlnx117:        at java.util.ComparableTimSort.sort(ComparableTimSort.java:146)
[2016-04-14T12:47Z] ukapdlnx117:        at java.util.Arrays.sort(Arrays.java:472)
[2016-04-14T12:47Z] ukapdlnx117:        at java.util.Collections.sort(Collections.java:155)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.outputFormatter.VcfOutputFormatter.addInfo(VcfOutputFormatter.java:78)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.outputFormatter.VcfOutputFormatter.toString(VcfOutputFormatter.java:285)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.outputFormatter.OutputFormatter.endSection(OutputFormatter.java:112)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.outputFormatter.VcfOutputFormatter.endSection(VcfOutputFormatter.java:229)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.outputFormatter.OutputFormatter.printSection(OutputFormatter.java:145)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEffCmdEff.annotate(SnpEffCmdEff.java:278)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEffCmdEff.annotateVcf(SnpEffCmdEff.java:446)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEffCmdEff.annotate(SnpEffCmdEff.java:138)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEffCmdEff.run(SnpEffCmdEff.java:984)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEffCmdEff.run(SnpEffCmdEff.java:939)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEff.run(SnpEff.java:978)
[2016-04-14T12:47Z] ukapdlnx117:        at ca.mcgill.mcb.pcingola.snpEffect.commandLine.SnpEff.main(SnpEff.java:136)
```
Thanks,
Miika

**Comment by etal** [comment]
I'd like for CNVkit to work with snpEff -- hopefully @pcingola can comment.

This VCF line looks pretty innocuous; can you tell which aspect of it snpEff didn't like? Maybe the negative value for SVLEN, but I think that is what VCF 4.2 specifies for deletions.

**Comment by mjafin** [comment]
Thanks Eric. @pcingola I know you're super busy but any feedback here would be welcome. 

The above is all the output I get from snpEff

**Comment by pcingola** [comment]
Apologies for the long time to answer.
This was a problem in SnpEff (not in CNVKit). I think that it 
has been solved in SnpEff v4.3:

    $ java -Xmx4g -jar snpEff.jar hg19 z.vcf | tee z.ann.vcf
    ...
    chr22	35006324	.	N	<DEL>	.	.	IMPRECISE;SVTYPE=DEL;END=35746318;SVLEN=-739994;ANN=<DEL>|chromosome_number_variation|HIGH|||chromosome|22|||n.35006325_35746318del||||||,<DEL>|feature_ablation|HIGH|ISX|ISX|gene_variant|ISX|||n.35006325_35746318del||||||,<DEL>|feature_ablation|HIGH|LINC01399|LINC01399|gene_variant|LINC01399|||n.35746318_35006325del||||||,<DEL>|feature_ablation|HIGH|HMGXB4|HMGXB4|gene_variant|HMGXB4|||n.35006325_35746318del||||||,<DEL>|feature_ablation|HIGH|TOM1|TOM1|gene_variant|TOM1|||n.35006325_35746318del||||||,<DEL>|feature_ablation|HIGH|MIR3909|MIR3909|gene_variant|MIR3909|||n.35006325_35746318del||||||,<DEL>|feature_ablation|HIGH|MIR6069|MIR6069|gene_variant|MIR6069|||n.35746318_35006325del||||||,<DEL>|feature_ablation|HIGH|||||||||||||,<DEL>|transcript_ablation|HIGH|ISX|ISX|transcript|NM_001303508.1|protein_coding|5/5|c.-456756_*264632del|p.0?|||||,<DEL>|transcript_ablation|HIGH|LINC01399|LINC01399|transcript|NR_126356.1|pseudogene|2/6|n.-119269_*509492del|p.0?|||||,<DEL>|transcript_ablation|HIGH|HMGXB4|HMGXB4|transcript|NM_001003681.2|protein_coding|7/11|c.-652059_*56674del|p.0?|||||,<DEL>|transcript_ablation|HIGH|TOM1|TOM1|transcript|NM_001135732.1|protein_coding|2/15|c.-689597_*3116del|p.0?|||||,<DEL>|transcript_ablation|HIGH|MIR3909|MIR3909|transcript|NR_037471.1|pseudogene|1/1|n.-725308_*14567del|p.0?|||||,<DEL>|transcript_ablation|HIGH|MIR6069|MIR6069|transcript|NR_106717.1|pseudogene|1/1|n.-13526_*726389del|p.0?|||||,<DEL>|transcript_ablation|HIGH|HMGXB4|HMGXB4|transcript|NR_027780.1|pseudogene|2/12|n.-647120_*54518del|p.0?|||||,<DEL>|transcript_ablation|HIGH|TOM1|TOM1|transcript|NM_001135729.1|protein_coding|15/15|c.-707592_*3116del|p.0?|||||,<DEL>|transcript_ablation|HIGH|TOM1|TOM1|transcript|NM_001135730.1|protein_coding|1/14|c.-689597_*3116del|p.0?|||||,<DEL>|transcript_ablation|HIGH|TOM1|TOM1|transcript|NM_005488.2|protein_coding|5/15|c.-689597_*3116del|p.0?|||||,<DEL>|transcript_ablation|HIGH|TOM1|TOM1|transcript|NR_024194.1|pseudogene|8/15|n.-689472_*2331del|p.0?|||||,<DEL>|transcript_ablation|HIGH|TOM1|TOM1|transcript|NR_024195.1|pseudogene|11/14|n.-689472_*2331del|p.0?|||||;LOF=(HMGXB4|HMGXB4|2|0.50),(ISX|ISX|1|1.00),(TOM1|TOM1|6|0.67)	GT:GQ	1/1:956

**Comment by etal** [comment]
Awesome. Thanks @pcingola and @mjafin !
